### PR TITLE
Add memory consumption estimation for models in profile API.

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
Add CPU/GPU memory consumption estimation for DL models in profile API. An example is as following,

{
  "nodes" : {
    "jJA5JA5ES1CG71Yugpc84g" : {
      "models" : {
        "FMtWfYcB6ZKIXgDcq6sw" : {
          "model_state" : "DEPLOYED",
          "predictor" : "org.opensearch.ml.engine.algorithms.text_embedding.TextEmbeddingModel@253a6c05",
          "target_worker_nodes" : [
            "jJA5JA5ES1CG71Yugpc84g"
          ],
          "worker_nodes" : [
            "jJA5JA5ES1CG71Yugpc84g"
          ],
          "mem_size_estimation_cpu" : 105529143,
          "mem_size_estimation_gpu" : 105529143
        },
        "p1hsfIcBSHStRf2jiLu6" : {
          "model_state" : "DEPLOYED",
          "predictor" : "org.opensearch.ml.engine.algorithms.text_embedding.TextEmbeddingModel@76d713a3",
          "target_worker_nodes" : [
            "jJA5JA5ES1CG71Yugpc84g"
          ],
          "worker_nodes" : [
            "jJA5JA5ES1CG71Yugpc84g"
          ],
          "mem_size_estimation_cpu" : 148025802,
          "mem_size_estimation_gpu" : 148025802
        }
      }
    }
  }
}
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java [hunk 1]
```diff
@@ -65,6 +66,59 @@ public synchronized void setModelState(String modelId, MLModelState state) {
...
+    /**
+     * Set memory size estimation CPU/GPU
+     * @param modelId model id
+     * @param format model format like onnx
+     * @param size memory size
+     */
+    public synchronized void setMemSizeEstimation(String modelId, MLModelFormat format, Long size) {
+        Long memSize = getMemSizeEstimation(format, size);
+        log.debug("Updating memSizeEstimation of Model {}  to {}", modelId, memSize);
+        getExistingModelCache(modelId).setMemSizeEstimationCPU(memSize);
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:78]: I see we set same value for CPU and GPU, does that mean the CPU and GPU memory consumption is almost the same?

File: plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java [hunk 2]
```diff
@@ -65,6 +66,59 @@ public synchronized void setModelState(String modelId, MLModelState state) {
...
+        getExistingModelCache(modelId).setMemSizeEstimationGPU(memSize);
+    }
+
+    private Long getMemSizeEstimation(MLModelFormat format, Long size) {
+        Double scale = 1.0;
+        switch (format) {
+            case ONNX:
+                scale = 1.5;
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:86]: Can we add a comment about explanation of setting up these magic numbers?

File: plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java (hunk 1 above)
Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:78]: I also have the same question. Github should have a `+1` icon :)

Comment (jngz-es) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:78]: Yes, from my experiments the cpu, gpu memory consumption is similar.

**Review by ylwu-amzn — APPROVED** [review]

**Review by dhrubo-os — DISMISSED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**